internal/characters/neuvillette/charge.go

```diff
@@ -0,0 +1,289 @@
+				ICDGroup:   attacks.ICDGroupDefault,
+				StrikeType: attacks.StrikeTypeDefault,
+				Element:    attributes.Hydro,
+				Durability: 25,
+				Mult:       charge[c.TalentLvlAttack()],
+			}
+			ap := combat.NewBoxHit(c.Core.Combat.Player(), c.Core.Combat.PrimaryTarget(), geometry.Point{}, 3, 8)
+			// TODO: Not sure of snapshot timing
+			c.Core.QueueAttack(
+				ai,
+				ap,
+				shortChargeHitmark+windup,
+				shortChargeHitmark+windup,
+			)
+		}
+	}, windup+3)
+
+	return action.Info{
+		Frames:          func(next action.Action) int { return windup + chargeFrames[next] },
+		AnimationLength: windup + chargeFrames[action.InvalidAction],
+		CanQueueAfter:   windup + chargeFrames[action.ActionDash],
+		State:           action.ChargeAttackState,
+	}, nil
+}
+
+func (c *char) judgementWave() {
+	// calculated every hit since canqueueafter is after the first tick, so configs can change the primary target/entity positions while the CA happens
+	ap := combat.NewBoxHit(c.Core.Combat.Player(), c.Core.Combat.PrimaryTarget(), geometry.Point{}, 3.5, 15)
+	if c.Base.Ascension >= 1 {
+		c.chargeAi.FlatDmg = chargeJudgement[c.TalentLvlAttack()] * c.MaxHP() * a1Multipliers[c.countA1()]
+	}
+	c.Core.QueueAttack(c.chargeAi, ap, 0, 0)
+}
+
+func getChargeJudgementHitmarkDelay(tick int) int {
+	// first tick happens 6f after start, second tick is 22f after first, then other frames are 25f after, then last tick is when the judgement wave ends.
+	// TODO: check this is the case for c6
+	switch tick {
+	case 0:
+		return 6
+	case 1:
+		return 22
+	default:
+		return 25
+	}
+}
+
+func (c *char) chargeJudgementTick(src, tick, maxTick int, last bool) func() {
+	return func() {
+		if c.chargeJudgeStartF != src {
+			return
+		}
+		// The charge attack has ended
+		if c.Core.F > c.chargeJudgeStartF+c.chargeJudgeDur {
+			return
+		}
+
+		if last {
+			if c.Core.F == c.chargeJudgeStartF+c.chargeJudgeDur {
+				c.judgementWave()
+			} else {
+				// c6 extended the CA between when this tick was queued and when this tick was executed
+				// allow the other non last queued task execute
+				c.nextTickTime = c.nextTickTime2
+			}
+			return
+		}
+
+		if tick == maxTick {
+			c.judgementWave()
+			c.chargeEarlyCancelled = true
+			return
+		}
+
+		c.judgementWave()
+		if maxTick == -1 || tick < maxTick {
+			nextF := c.Core.F + getChargeJudgementHitmarkDelay(tick+1)
+
+			// this one is always queued. We don't cancel this even if it's beyond the CA duration in case the c6 extends the charge attack between this tick and the "final" tick
+			c.QueueCharTask(c.chargeJudgementTick(src, tick+1, maxTick, false), nextF-c.Core.F)
+			c.nextTickTime = nextF - c.chargeJudgeStartF
+
+			if nextF > c.chargeJudgeStartF+c.chargeJudgeDur {
+				c.QueueCharTask(c.chargeJudgementTick(src, tick+1, maxTick, true), c.chargeJudgeStartF+c.chargeJudgeDur-c.Core.F)
+				c.nextTickTime = c.chargeJudgeDur
+				c.nextTickTime2 = nextF - c.chargeJudgeStartF
+			}
+		}
+	}
+}
```

Comment:
Please change the implementation to:
```
func (c *char) chargeJudgementTick(src, tick, maxTick int, last bool) func() {
	return func() {
		if c.chargeJudgeStartF != src {
			return
		}
		// no longer in CA anim -> no tick
		if c.Core.F > c.chargeJudgeStartF+c.chargeJudgeDur {
			return
		}

		// last tick -> check for C6 extension
		if last {
			// C6 did not extend CA -> proc wave and stop queuing ticks
			if c.Core.F == c.chargeJudgeStartF+c.chargeJudgeDur {
				c.judgementWave()
			} else {
				// C6 extended the CA between when this tick was queued and when this tick was executed
				// -> allow the other non last queued task to execute by extend anim length to include that task
				c.tickAnimLength = c.tickAnimLengthC6Extend
			}
			return
		}

		// tick param supplied and hit the limit -> proc wave, enable early cancel flag for next action check and stop queuing ticks
		if tick == maxTick {
			c.judgementWave()
			c.chargeEarlyCancelled = true
			return
		}

		c.judgementWave()

		// next tick handling
		if maxTick == -1 || tick < maxTick {
			tickDelay := getChargeJudgementHitmarkDelay(tick + 1)
			// calc new animation length to be up until next tick happens
			nextTickAnimLength := c.Core.F - c.chargeJudgeStartF + tickDelay

			// always queue up non-final tick that will be executed in case C6 was proc'd
			c.Core.Tasks.Add(c.chargeJudgementTick(src, tick+1, maxTick, false), tickDelay)

			// queue up last tick if next tick would happen after CA duration ends
			if nextTickAnimLength > c.chargeJudgeDur {
				// queue up final tick to happen at end of CA duration
				c.Core.Tasks.Add(c.chargeJudgementTick(src, tick+1, maxTick, true), c.chargeJudgeDur-c.tickAnimLength)
				// update tickAnimLength to be equal to entire CA duration at the end
				c.tickAnimLength = c.chargeJudgeDur
				// if C6 is triggered, then tickAnimLength will be wrong so this var holds the actual tickAnimLength if ticks continued normally beyond the original final tick
				c.tickAnimLengthC6Extend = nextTickAnimLength
			} else {
				// next tick happens within CA duration -> update tickAnimLength as usual
				c.tickAnimLength = nextTickAnimLength
			}
		}
	}
}
```
This + the suggested change for the very first queuing of the task should fix the bug when ticks param < 8 the actual ticks are 1 more than supplied.
Includes:
- nextTickTime -> tickAnimLength
- nextTimeTime2 -> tickAnimLengthC6Extend
- bunch of comments
- less frame arithmetic / simpler conditions for the next tick part at the bottom
- no char queue because skill aligned hitlag leads to desync (see change suggested for very tick for reasoning)

I tested it out a bit (non-c6 and ticks < 8, non-c6 and no ticks param, c6 and ticks <8, c6 and no ticks param, c6 and ticks param > 8) and it seems to work well.

Reply:
I additionally updated the `getChargeJudgementHitmarkDelay` function to match the 1 indexed tick counting, and correspondingly updated the initial call to use getDelay(1)